samples/client/petstore/rust-reqwest/src/models/api_response.rs

```diff
@@ -0,0 +1,90 @@
+/* 
+ * OpenAPI Petstore
+ *
+ * This is a sample server Petstore server. For this sample, you can use the api key `special-key` to test the authorization filters.
+ *
+ * OpenAPI spec version: 1.0.0
+ * 
+ * Generated by: https://openapi-generator.tech
+ */
+
+/// ApiResponse : Describes the result of uploading an image resource
+
+#[allow(unused_imports)]
+use serde_json::Value;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ApiResponse {
+  #[serde(rename = "code")]
+  code: Option<i32>,
+  #[serde(rename = "type")]
+  _type: Option<String>,
+  #[serde(rename = "message")]
+  message: Option<String>
+}
+
+impl ApiResponse {
+  /// Describes the result of uploading an image resource
+  pub fn new() -> ApiResponse {
+    ApiResponse {
+      code: None,
+      _type: None,
+      message: None
+    }
```

Comment:
Can you not implement default instead?

Reply:
After a closer look, it is not as simple. When there is a required var in a model, default constructor `new` has required vars as parameters. So, `Default` trait cannot be implemented in this case.

Reply:
Good point - https://github.com/nrc/derive-new might be what you want, then.